examples/methods/ada_svr.py

```diff
@@ -0,0 +1,203 @@
+#  :alt: Figure 1 of ::footcite:ct:`Gaonkar et al. 2012 <gaonkar_deriving_2012>`.
+#  :align: center
+#  :name: fig:ada_svr_1
+#
+#  Figure 1 of :footcite:ct:`Gaonkar et al. 2012 <gaonkar_deriving_2012>`.
+#  **Left**: Concept of support vector machines in 2-D space
+#  **Right**: Permutation testing for support vector machines
+#
+
+#############################################################################
+# Comparison with the permutation test for SVR for validating the approach
+# ------------------------------------------------------------------------
+#
+estimator = SVR(kernel="linear", epsilon=0.0, gamma="scale", C=1.0)
+estimator.fit(X, y)
+beta_hat_svr = estimator.coef_
+
+# compare that the coefficiants are the same that the one of SVR
+assert np.max(np.abs(beta_hat - beta_hat_svr.T[:, 0])) < 1e-4
+
+#############################################################################
+# This coefficient of SVR is an estimation of the importance of variables.
+# For estimating the confidence interval of the importance of variables,
+# ADA-SVR uses propose to estimate the distribution of the coefficients
+# of SVR, instead of using the classical permutation test for this estimation.
+
+#############################################################################
+# Estimation of the distribution of the coefficients of SVR
+# ---------------------------------------------------------
+#
+
+proba = permutation_test(
+    X, y, estimator=estimator, n_permutations=10000, n_jobs=8, seed=42, proba=True
+)
+plot_compare_proba_estimated(proba, beta_hat, scale)
+
+#############################################################################
+# **Compare the distribution of the coefficients of SVR with the
+# estimation of the distribution of the coefficients by ADA-SVR**
+print(
+    "ADA-SVR assumes that the normal distribution of the coefficient is",
+    "center around zeros.\n",
+    "Our estimation is that the maximum deviation is: {:.4f}\n".format(
+        np.max(np.abs(np.mean(proba, axis=0)))
+    ),
+    "ADA-SVR provides that the standard deviation the normal distribution",
+    "for each coefficients.\n",
+    "The maximum of difference between AdA-SVR estimation and our estimation",
+    "is:{:.4f}".format(np.max(np.abs(scale - np.std(proba, axis=0)) / scale)),
+)
+
+#############################################################################
+# Assumptions, Advantages and Disadvantages
+# -----------------------------------------
+#
+# **Assumptions**:
+#
+# - The distribution of the coefficients of SVR is normal centred around zeros.
+# - The method is valid for large sample sizes.
+# - The method has the linear models assumptions: linearity, normality, 
+#   homoscedasticity, independence, fixed features, absence of multicollinearity
+#   (see the book of :footcite:ct:`Molnar 2012<molnar2020interpretable>`
+#   for details)
+#
+# **Advantages**:
+# 
+# - The method is fast because it uses the central limit theorem to estimate
```

Comment:
It is actually fast because it uses linear regression to get the uncertainty estimate.

Reply:
the CLT is simply there to strengthen some Gaussian assumption. It is here for the sake of theory only.

Reply:
We can say it's fast because of it's a broad estimate of the weight distribution.